zerver/lib/email_notifications.py

```diff
@@ -372,16 +368,13 @@ def message_header(message: Message) -> dict[str, Any]:
     #    },
     # ]
 
+    assert len(messages) > 0
+    header = message_header(messages[0])
     messages.sort(key=lambda message: message.date_sent)
 
     for message in messages:
-        header = message_header(message)
```

Comment:
I feel like if we're making this change, `build_message_list` docstrings need to change. Previously, this supported an arbitrary feed of messages, potentially with multiple headers. We're not using that capability, so it may be fine to remove. But we should document (and `assert` that all messages have the same recipient) if we're doing to make the code wrong in the case that assumption was wrong.